test/performance/long-running-tests/README.md

```diff
@@ -0,0 +1,174 @@
+
+
+4. Deploy your first workload:
+
+   First, use a kubeconfig that includes contexts for WDS spaces and set its current context to a target WDS space (e.g., `wds1`):
+
+   ```bash 
+   kubectl config use-context wds1
+   ```
+
+   Output:
+   ```console 
+   Switched to context "wds1".
+   ```
+   
+   Optionally, check the kubeconfig contexts: 
+   ```bash 
+   kubectl config get-contexts
+   ```
+   
+   Output:
+   ```console 
+   CURRENT   NAME            CLUSTER         AUTHINFO        NAMESPACE
+             cluster1        kind-cluster1   kind-cluster1   
+             cluster2        kind-cluster2   kind-cluster2   
+             its1            its1-cluster    its1-admin      
+             kind-kubeflex   kind-kubeflex   kind-kubeflex   
+   *         wds1            wds1-cluster    wds1-admin      default
+             wds2            wds2-cluster    wds2-admin      default
+   ```
+
+   Second, create the binding Policy and namespace for your experiment using the following commands:
+
+   ```bash 
+   cd $CL2_DIR/testing/load/
+   kubectl apply -f long-duration-exp-setup.yaml
+   ```
+
+   Lastly, in the following set ``--kubeconfig`` flag to the path of the kubeconfig file used above and use the newly added Kubestellar provider (`--provider=ks`) and run the following command to create the workload:
+
+   ```bash
+   cd $CL2_DIR
+   go run cmd/clusterloader.go --testconfig=./testing/load/long-duration-exp-config.yaml --kubeconfig=<path>/wds-kubeconfig --provider=ks --v=2
+   ```
+    
+   The generated workload is a pod that sleeps for 20 seconds([see here](workloads/long-duration-exp-workload.yaml)) and created using the clusterloader `SteppedLoad` tuningSet (burstSize=1 and stepDelay=60 sec). SteppedLoad defines a load that generates a burst of a given size every X seconds, see more [here](https://github.com/kubernetes/perf-tests/blob/6ec633314941e74d8d8e31c83ac8711e7c3d9832/clusterloader2/docs/design.md#tuning-set).
+
+
+5. Run the metrics collection script: collect the creation and status_update timestamps for the benchmark workload objects proceed as following
+
+   a) cd into the `test/performance/common` directory from your local copy of the KubeStellar repo, for example:
+
+   ```bash 
+   cd .$HOME/kubestellar/test/performance/common
+   ```
+
+   b) Run the metrics collection script:
+
+   ```bash
+   python3 metrics_collector.py <kubeconfig> <wds-context-name> <its-context-name> <wec-context-name> <number-of-namespaces> <output-directory> <exp-type> <number-of-pods> <watch-interval>
+   ```
+
+   For example:
+   
+   ```bash 
+   python3 metrics_collector.py $HOME/.kube/config wds1 its1 cluster1 1 $HOME/data l 10 30
+   ```
+
+   Observations: 
+      - The execution of the above python script requires the installation of the dependencies listed in `requirements.txt`.
```

Comment:
FYI, it is more reader-friendly to mention prerequisites before exhibiting the command that depends on those prerequisites.

FYI, the typical practice for a Python user is to make a virtual environment for each project/app/directory/whatever that has its own requirements.txt. Typically that goes in a directory in that project/app/directory/whatever. When that is under Git control, this presents a little pain. I think that best practice is for you to recommend a specific place to put that venv and add that place to the project's `.gitignore`.

Reply:
IMO, the user may already have its own virtual environment set-up in his environment that he could  just re-use. I believe that it's beyond the scope of these instructions to provide guidance on how the user should set-up his python virtual environment. Personally, I believe that env should be set-up outside of any git directory.

Reply:
@MikeSpreitzer, I moved the text about the virtual environment into the list of prerequisites here: https://github.com/dumb0002/kubestellar/tree/long-duration-exp/test/performance#readme.

Reply:
OK, the added text about prerequisites is good.

BTW, I am not trying to suggest that we duplicate the instructions for how to make a Python virtual environment. My suggestion is that this document recommend a particular place for that virtual environment to appear and add that place to `.gitignore`. Since the `requirements.txt` is in `test/performance/common/`, I think it would be natural for the venv to appear there too. In particular, I think the most likely recommendations are `test/performance/common/.venv` or `test/performance/common/venv`.

Reply:
Ok, I updated the instructions here with the relevant information: https://github.com/dumb0002/kubestellar/tree/long-duration-exp/test/performance#readme